lib/ui/painting.dart

```diff
@@ -1830,6 +1835,49 @@ class Image {
   String toString() => _image.toString();
 }
 
+class ImagePerfData {
+  ImagePerfData(this.resourceDecodeTime);
+  int resourceDecodeTime;
+}
+
+// ignore: avoid_classes_with_only_static_members
+class ImagePerfInfos {
+  static Expando<ImagePerfData> expando = Expando<ImagePerfData>('ImagePerfInfos');
+  static int getResourceDecodeTime(Image image) {
+    final ImagePerfData? info = expando[image];
+    if (info == null) {
+      return 0;
+    }
+    return info.resourceDecodeTime;
+  }
+  static void setResourceDecodeTime(Image image, int time) {
+    expando[image] = ImagePerfData(time);
+  }
+  static void setResourceDecodeTimePri(_Image image, int time) {
+    if (!kReleaseMode) {
+      image.resourceDecodeTime = time;
+      for(final Image imageh in image.handles) {
+        setResourceDecodeTime(imageh, time);
+      }
+    }
+  }
+}
+
+extension ImagePerfInfo on Image {
+  int get resourceDecodeTime {
+    return ImagePerfInfos.getResourceDecodeTime(this);
+  }
+
+  set resourceDecodeTime(int time) {
```

Comment:
This property needs documentation that indicates it is not available in profile mode.

Reply:
How can we preserve this function in profile mode? Maybe with Pragma vm:entry-point ?

Reply:
@chinmaygarde any suggestions? Are there any existing similar cases in the engine? Another option could be to store the data on the C++ side so that C++ does not need to call back into dart to store the decode time.